### PR TITLE
Allow to change _START _END tokens

### PR DESCRIPTION
**Pull request by ghost**
Hi,
we love Plush very much, we use it in our open source project https://oya.sh/ as a engine for rendering.
It would very helpful to be able to change starting(S_START, C_START, E_START) and ending(E_END) tokens to something else. 
I came up with something like this allowing user to change those tokens with `token.SetTemplatingDelimiters(start, end)` and fetch them with `token.Resolve`.
 I had quite a problem with naming convention in lexer so feel free to point out everything you see. thanks

### CODE REVIEW

**Comment by b4rt84ek** [comment]
> Found some fixes!

all right! ;)

File: token/token.go [hunk 1]
```diff
@@ -23,10 +36,37 @@ var keywords = map[string]Type{
...
 	return IDENT
 }
+
+// SetTemplatingDelimiters to start and end, or return delimitersLengthError if delimiters length is incorrect
+func SetTemplatingDelimiters(start, end string) error {
+	if len(start) != templateDelimitersLen ||
+		len(end) != templateDelimitersLen {
+		return &delimitersLengthError{[]string{start, end}, templateDelimitersLen}
+	}
+	replace(S_START, Type(start))
```
Comment (bilus) [inline token/token.go]: Does the `replace` function really add value in comparison to a simple assignment?

File: lexer/lexer.go [hunk 2]
```diff
@@ -49,11 +50,46 @@ func (l *Lexer) NextToken() token.Token {
 	return tok
 }
 
+func (l *Lexer) nextOutsideToken() (token.Token, bool) {
+	var tok token.Token
```
[the remark this answers is not in the document]
Comment (bart84ek) [reply lexer/lexer.go]: indeed, thanks

File: token/token.go (hunk 1 above)
Comment (bart84ek) [reply token/token.go]: agree, fixed

**Comment by paganotoni** [comment]
Closing because of no recent activity.

**Pull request closed without merging**